### ISSUE TITLE
Set frame of texture in particle2d

### ISSUE THREAD

**Issue by ondesic** [godotengine/godot#30995]
In 3.1, Godot has moved the animated frames for a texture to CanvasItemMaterial. I can set the amount of frames, but would like to set which frame of the texture is showing. Is there a `set_particle_frame` or something similar that I can use to manually change the particle frame with code?  If not this would be a good addition

**Comment by KoBeWi** [comment]
In `ParticlesMaterial` under `Animation` there is a property called `offset`, which controls which frame is displayed. Offset of 0 means first frame and 1 means last frame.

Next time if you are unsure if a feature exists, [ask on other community channels](https://godotengine.org/community) before requesting them here.

**Comment by ondesic** [comment]
KoBeWi,

Thank you for responding.  I actually did ask this question in another community and no one knew.(https://godotengine.org/qa/49350/set-frame-of-texture-in-particle2d)  . 
I am ultra careful to check whether a question has already been asked.

Please don't close this yet, I feel that for continuity across Godot that similar features should be called the same.  "Offset" really could be more descriptive.  Can we name it "frame_offset" or include a "frame" property?

The user knows how many frames are in their texture.  It would be very helpful to have an actual "frame" property.  Having a property for frames called "offset" range from 0 to 1 isn't helpful when you have 13 frames and want to pick the 4th frame :)

**Comment by KoBeWi** [comment]
Ok, reopened per request. Either we can rename the property or improve the documentation, since the docs don't say what it really does or how to use it.

Also
> Having a property for frames called "offset" ranging from 0 to 1 isn't helpful when you have 13 frames and want to pick the 4th frame :)

You can just do 4.0/13. It works even in the inspector.

**Comment by ondesic** [comment]
yes 4.0/13 would work, usually there is always a work around.   I think for someone use to selecting frames from a texture, a frame property would be great and intuitive.  Thank you for the discussion and thoughts
